### DISCUSSION TITLE
Fritzing diagram or photos of circuit

### DISCUSSION THREAD

**Discussion by ryanalexmartin** [corrados/edrumulus#88] [General]
![image](https://github.com/corrados/edrumulus/assets/15975303/03ea0b1d-8a25-4341-96f5-3956d409befa)

I've tried to build this circuit a few times, but I think I might be reading it incorrectly.  (Not an electrical engineer)

Does anybody have a Fritzing diagram of the basic circuit shown in the README?  Or perhaps a photo of a breadboard/perfboard?

If not, in the interest of reducing the work I'm asking of others by posting this question, could somebody perhaps take a look at this Fritzing diagram I made and check if I have anything wired incorrectly?

This is wired for only one drum, no rim shot, and no hi-hat controller (so not exactly equal to the above schematic) and the wires on the left are 3.3V, GND, and the snare drum input pin.  I don't have the ESP32 part, so I just connected them to nothing.

<img width="776" alt="Screenshot 2023-08-16 at 7 18 14 PM" src="https://github.com/corrados/edrumulus/assets/15975303/eff6f58f-6dc7-4735-919a-e41ee2ff1f08">

I did notice that I used an incorrect resistor value the last time I built the circuit.  I'll head to the electronics shop and purchase some 22k ohm resistors.  Looks like I was mistakenly using 10k ohms last time I attempted to make the circuit.

**Comment by corrados** [comment]
> take a look at this Fritzing diagram I made and check if I have anything wired incorrectly?

Looks correct to me.

**Comment by ryanalexmartin** [comment]
<img width="864" alt="Screenshot 2023-08-17 at 7 37 04 PM" src="https://github.com/corrados/edrumulus/assets/15975303/dd770aa8-2d61-432b-98d6-e93ae4fb2560">

With the circuit wired up as above, I'm getting this noise when the jumper is placed on the specified pin.  Here's the analogRead() value of the pin in question, just a bunch of noise with values hovering around 2000.


<img width="882" alt="Screenshot 2023-08-17 at 7 39 17 PM" src="https://github.com/corrados/edrumulus/assets/15975303/3c5f5802-798e-47c9-ab4f-dd05f7321f26">
When I remove the pin in question, I get a kind of compound sine wave.

I've tried a few different piezos, a few different cables, and double checked all the components to make sure they're not faulty.

Also, perhaps it's worth noting that presence of the 100k ohm resistor to the circuit does not affect the pin voltage.

Also, I should mention that plugging the piezo into the 3.5mm jack does not affect the noise shown above.  Seems the circuit is in a state of floating ground?

**Reply by corrados** [reply]
> With the circuit wired up as above, I'm getting this noise when the jumper is placed on the specified pin. Here's the analogRead() value of the pin in question, just a bunch of noise with values hovering around 2000.

That's perfectly ok. It looks the same for my prototypes as well.

> Also, perhaps it's worth noting that presence of the 100k ohm resistor to the circuit does not affect the pin voltage.

Yes, this resistor is only to lower the voltage which the ADC of the ESP32 sees.

> Also, I should mention that plugging the piezo into the 3.5mm jack does not affect the noise shown above. Seems the circuit is in a state of floating ground?

I would actually not expect the noise to change when a piezo is connected.

**Comment by ryanalexmartin** [comment]
![IMG_1191](https://github.com/corrados/edrumulus/assets/15975303/3ac64226-a8f7-4e58-9822-d1ed03a695c4)
![IMG_1190](https://github.com/corrados/edrumulus/assets/15975303/312f4463-2aee-4039-a569-09e6042dde5d)

I see two potential possibilities.

1.) Something in my breadboard circuit is not matching the above photographs, or I have some misunderstanding in the way things should be wired.

2.). Something could have gone wrong with my dev environment.  I've seen in the changelog that there are some versions of the ESP32 Arduino package that have caused some issues.  I noticed that my Arduino IDE updated to 2.0.11 without my permission.  Is version 1.0.6 still recommended?  Trying to compile with version 1.0.6 gives the following error:

```
/Users/ryan/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-97-gc752ad5-5.2.0/xtensa-esp32-elf/include/c++/5.2.0/bits/stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
/Users/ryan/git/edrumulus/edrumulus.cpp:574:102: note:   deduced conflicting types for parameter 'const _Tp' ('float' and 'double')
                                                         ( i * pad_settings.clip_comp_ampmap_step ) ) );
                                                                                                      ^

exit status 1

Compilation error: no matching function for call to 'min(float, double)'
```

**Reply by corrados** [reply]
> 1.) Something in my breadboard circuit is not matching the above photographs, or I have some misunderstanding in the way things should be wired.

It looks ok to me...

> I noticed that my Arduino IDE updated to 2.0.11 without my permission. Is version 1.0.6 still recommended?

I am also using version 2 and it works fine for me. You need to make sure that the ESP32 library has the correct version. I am using Arduino IDE version 2.1.0 but your version 2.0.11 should also work fine:
![grafik](https://github.com/corrados/edrumulus/assets/46655886/075584ff-30b8-495e-bb7d-975ae9015584)
